panther_lights/src/controller_node.py

```diff
@@ -112,14 +189,30 @@ def _percent_to_apa_driver_brightness(self, brightness_percent) -> int:
 
     def _controller_timer_cb(self, *args) -> None:
         with self._lock:
-            if self._animation_finished or self._interrupt:
-                if self._anim_queue.empty():
-                    self._current_animation = self._default_animation
-                else:
-                    self._current_animation = self._anim_queue.get(block=False)
-                    self._interrupt = False
+            if self._animation_finished:
+                self._anim_queue.validate_queue()
+
+                if self._default_animation and not self._anim_queue.has_animation(
+                    self._default_animation
+                ):
```

Comment:
```suggestion
                if self._default_animation and not
                    self._anim_queue.has_animation(self._default_animation):
```

This is more of a question if this isn't more readable

Reply:
black auto formatting

Reply:
k